back-nest/src/products/products.services.ts

```diff
@@ -64,4 +64,21 @@ export class ProductService {
       }),
     );
   }
+
+  async deleteProduct(id: number): Promise<void> {
+    const product = await this.prisma.product.findUnique({
+      where: { id },
+    });
+
+    if (!product) {
+      throw new NotFoundException(`Product with id ${id} not found`);
+    }
```

Comment:
Prisma [delete](https://www.prisma.io/docs/orm/reference/prisma-client-reference#delete) already checks that. If record doesn't exist, Prisma will throw an exception automatically